kernel.go

```diff
@@ -373,6 +379,82 @@ func handleExecuteRequest(ir *classic.Interp, receipt msgReceipt) error {
 	return receipt.Reply("execute_reply", content)
 }
 
+// doEval evaluates the code in the interpreter. This function captures an uncaught panic
+// as well as the values of the last statement/expression.
+func doEval(ir *classic.Interp, code string) (_ []interface{}, err error) {
+	// Capture a panic from the evaluation if one occurs and store it in the `err` return parameter.
+	defer func() {
+		if r := recover(); r != nil {
+			var ok bool
```

Comment:
Maybe just:

```
			if err, ok := r.(error); !ok {
				err = errors.New(fmt.Sprint(r))
			}
```

Reply:
I know it doesn't look pretty but the `ok` needs to be declared first so that the `:=` assignment is not used to make `err` refer to the named return and not create a local variable.